kart/point_cloud/v1.py

```diff
@@ -84,13 +102,41 @@ def diff_tile(self, other, tile_filter=FeatureKeyFilter.MATCH_ALL, reverse=False
             "tile",
             key_filter=tile_filter,
             key_decoder_method="decode_tile_path",
-            value_decoder_method="get_tile_summary_from_pointer_blob",
+            value_decoder_method="get_tile_summary_promise_from_path",
             reverse=reverse,
         )
 
-    def get_tile_summary_from_pointer_blob(self, tile_pointer_blob):
-        # For now just return the blob contents as a string - it's a reasonable summary.
-        result = pointer_file_to_json(tile_pointer_blob)
-        if "version" in result:
-            del result["version"]
-        return result
+    def get_tile_summary_promise_from_path(self, tile_path):
+        tile_pointer_blob = self.get_blob_at(tile_path)
+        return functools.partial(
+            self.get_tile_summary_from_pointer_blob, tile_pointer_blob
+        )
+
+    def diff_to_wc(self, wc_diff_context, ds_filter=DatasetKeyFilter.MATCH_ALL):
+        ds_diff = DatasetDiff()
+        tile_filter = ds_filter.get("tile", ds_filter.child_type())
+        ds_diff["tile"] = DeltaDiff(self.diff_tile_to_wc(wc_diff_context, tile_filter))
+        return ds_diff
+
+    def diff_tile_to_wc(self, wc_diff_context, tile_filter):
+        deltas = wc_diff_context.workdir_deltas_by_ds_path().get(self.path)
+        if not deltas:
+            return
+
+        def encode_tile_path_relative(tilename):
+            return self.encode_tile_path(tilename, relative=True)
+
+        yield from self.decode_raw_deltas(
+            deltas,
+            tile_filter,
+            old_path_decoder=encode_tile_path_relative,
```

Comment:
might be a terminology conflict here since you're passing an `encode_` function to kwarg called `_decoder`, if that's correct them maybe it  needs a comment? Or rename something :)

Reply:
Yeah this got a bit weird. Will tidy up some names and add some comments.

(Vector datasets have paths which *encode* to primary keys, and *decode* back to paths again. Vector encoder / decoder doesn't deal with the working copy, which doesn't encode primary keys, and is in a separate part of Kart.
Whereas: for point cloud datasets, tiles have complicated paths (for reasons of limiting tree-size)... which... decode(?) to their names? And encode back to their complicated paths? But also, in the working copy, tiles have a different path... the tidy encode / decode system started breaking down a bit)